modules/querier/querier.go

```diff
@@ -82,34 +83,49 @@ func New(cfg Config, clientCfg ingester_client.Config, ring ring.ReadRing, store
 		limits: limits,
 	}
 
-	q.subservicesWatcher = services.NewFailureWatcher()
-	q.subservicesWatcher.WatchService(q.pool)
-
 	q.Service = services.NewBasicService(q.starting, q.running, q.stopping)
 	return q, nil
 }
 
+// todo: fix name, its confusing
+func (q *Querier) RegisterSubservices(w services.Service) error {
+	var err error
+	q.subservices, err = services.NewManager(w, q.pool)
+	q.subservicesWatcher = services.NewFailureWatcher()
+	q.subservicesWatcher.WatchManager(q.subservices)
+	return err
+}
+
 func (q *Querier) starting(ctx context.Context) error {
-	err := services.StartAndAwaitRunning(ctx, q.pool)
-	if err != nil {
-		return fmt.Errorf("failed to start pool %w", err)
+	if q.subservices != nil {
```

Comment:
I understand that the ergonomics of the services Manager and FailureWatcher are not great, but I really don't like checking for nil here.  Shouldn't we always have a subservices manager? for at least the ring client (`q.pool`)?

Same concerns in `running()` and `stopping()`